src/presentation/components/Convert/useConvert.ts

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from 'react';
+import {
+  useConfirmConvertBalanceMutation,
+  useConvertBalanceMutation,
+  useGetBalanceMutation,
+  useGetCoinsMutation,
+  useGetConvertBalanceMutation,
+} from 'infrastructure/services/deposit/DepositService';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
+
+interface FormValues {
+  fromCoin: string;
+  toCoin: string;
+  size: string;
+}
+
+export const useConvert = () => {
+  const [showResume, setShowResume] = useState(false);
+  const [isConvertExpired, setIsConvertExpired] = useState(true);
+  const [currencies, setCurrencies] = useState<string[]>([]);
+  const [error, setError] = useState('');
+
+  const [getCoins, { data: getCoinsData, isSuccess: isGetCoinSuccess }] = useGetCoinsMutation();
+  const [getBalance] = useGetBalanceMutation();
+  const [
+    convertBalance,
+    { isSuccess: isConvertBalanceSuccess, isError: isConvertBalanceError, data, error: convertBalanceError },
+  ] = useConvertBalanceMutation();
+  const [confirmConvertBalance, { isSuccess: isConfirmConvertBalanceSuccess, isLoading }] =
+    useConfirmConvertBalanceMutation();
+  const [getConvertBalance, { isSuccess: isGetConvertBalanceSuccess, data: getConvertData }] =
+    useGetConvertBalanceMutation();
+
+  const schema = z
+    .object({
+      fromCoin: z.string().min(3, { message: 'Field Required' }),
+      toCoin: z.string().min(3, { message: 'Field Required' }),
+      size: z.string().min(1, { message: 'Field Required' }),
+    })
+    .refine(data => data.fromCoin !== data.toCoin, {
+      message: 'Coins must be different',
+      path: ['fromCoin', 'toCoin'],
+    });
+
+  const {
+    register,
+    handleSubmit,
+    clearErrors,
+    reset,
+    formState: { errors },
+  } = useForm({ resolver: zodResolver(schema) });
+
+  const onSubmit: SubmitHandler<FormValues> = async data => {
+    await convertBalance(data);
+  };
+
+  useEffect(() => {
+    getCoins();
+  }, [getCoins]);
+
+  useEffect(() => {
+    if (isConvertBalanceSuccess) {
+      setError('');
+      getConvertBalance(data.result.quoteId);
+    }
+  }, [isConvertBalanceSuccess]);
+
+  useEffect(() => {
+    if (isGetCoinSuccess) {
+      const currencies = getCoinsData.result.map(coin => coin.id);
+      setCurrencies(currencies);
+    }
+  }, [isGetCoinSuccess]);
+
+  useEffect(() => {
+    if (isGetConvertBalanceSuccess) {
+      setIsConvertExpired(false);
+      setShowResume(true);
+
+      setTimeout(() => {
+        setShowResume(false);
+
+        setIsConvertExpired(true);
+      }, 4000);
```

Comment:
is it necessary to use a setTimeout function here?

Reply:
Yes, It's needed here because once you create a quote this has about 4 seconds of expiration time, So we need to disabled button after that time